shared/macros/10-bash.jinja

```diff
@@ -1875,40 +1875,133 @@ Part of the grub2_bootloader_argument_absent template.
-  If the package is installed, the Bash remediation will be applied.
-  The macro respects `platform_package_overrides` variable.
+This macro creates a bash conditional which is used to determine if a
+remediation is applicable. The macro takes package as an argument and chooses
+appropriate package manager. If the package is installed and satisfies the
+optional version restricion, the Bash remediation will be applied. The macro
+respects `platform_package_overrides` variable.
+
+:param package: package name
+:param ver: package version (optional argument, use together with "op")
+:param op: version comparison operator ("<", "<=", "==", "!=", ">", ">=")
 #}}
-{{%- macro bash_pkg_conditional(package, escape=False) -%}}
+{{%- macro bash_pkg_conditional(package, ver=None, op=None) -%}}
 {{%- if package in platform_package_overrides -%}}
   {{%- set package = platform_package_overrides[package] -%}}
 {{%- endif -%}}
   {{% if pkg_system is defined %}}
     {{%- if pkg_system == "rpm" -%}}
-        {{{ bash_pkg_conditional_rpm(package) }}}
+        {{{ bash_pkg_conditional_rpm(package, ver, op) }}}
     {{%- elif pkg_system == "dpkg" -%}}
-        {{{ bash_pkg_conditional_dpkg(package, escape) }}}
+        {{{ bash_pkg_conditional_dpkg(package, ver, op) }}}
     {{%- else -%}}
 JINJA MACRO ERROR - Unknown package system '{{{ pkg_system }}}'.
     {{%- endif -%}}
 {{% endif %}}
 {{%- endmacro -%}}
 
+{{#
+This macro generates code that gets version of an installed RPM package.
+
+:param package: package name
+#}}
+{{%- macro bash_get_rpm_package_version(package) -%}}
+rpm -q --queryformat '%{VERSION}-%{RELEASE}' {{{ package }}}
+{{%- endmacro -%}}
+
+{{#
+This macro creates a Bash conditional that compares version of the RPM package
+with a given version.
+
+Description of the algorithm:
+1. Get the actual version of the given package using the `rpm` command
+   and store it in `real`.
+2. Store the expected version in `ver`.
+3. Perform the comparison and return the result.
+   Comparison method is different based on the comparison operator. The method
+   code is chosen at the build time during Jinja expansion. Therefore, the
+   algorithm doesn't use the operator at all.
+   Based on the operator, these operations are performed:
+   a. "<": real != ver && is_sorted([real, ver])
+   b. "<=": is_sorted([real, ver])
+   c. "==": real == ver
+   d. "!=": real != ver
+   e. ">=" real != ver && is_sorted([ver, real])
+   f. ">" is_sorted([ver, real])
+   where is_sorted returns true if the given list parameter is a sorted list of version numbers.
+
+The implementation uses the GNU `sort` version ordering, which is described at:
+https://www.gnu.org/software/coreutils/manual/coreutils.html#Version-sort-ordering
+
+:param package: package name
+:param ver: package version
+:param op: version comparison operator ("<", "<=", "==", "!=", ">", ">=")
+#}}
+{{%- macro bash_compare_version_rpm(package, ver, op) -%}}
```

Comment:
You couple two things here - retrieval of the version and the comparison algorithm.
Why not to redesign it to have
1. A macro `bash_compare_version` that accepts a Bash string,
2. Wrapping it in a macro `bash_compare_version_rpm` that would call the former with `$({{{ bash_get_rpm_package_version(package) }}})` as an argument?

That way, you won't have to mock the `rpm` command in tests.